### PR TITLE
Apply nms update

### PR DESCRIPTION
**Pull request by harisreedhar**

### CODE REVIEW

File: facefusion/typing.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
-from typing import Any, Literal, Callable, List, Tuple, Dict, TypedDict, Optional
+from typing import Any, Literal, Callable, List, Tuple, Dict, TypedDict, Optional, Sequence
+from numpy.typing import NDArray
+from cv2.typing import Size
 from collections import namedtuple
 import numpy
 
-BoundingBox = numpy.ndarray[Any, Any]
+BoundingBox = NDArray[numpy.float32]
```
Comment (henryruhs) [inline facefusion/typing.py:7]: apply this to every `numpy.ndarray[Any, Any]` possible?

File: facefusion/face_helper.py [hunk 2]
```diff
@@ -136,28 +136,14 @@ def convert_face_landmark_68_to_5(face_landmark_68 : FaceLandmark68) -> FaceLand
...
-		yy1 = numpy.maximum(y1[index], y1[remain_indices])
-		xx2 = numpy.minimum(x2[index], x2[remain_indices])
-		yy2 = numpy.minimum(y2[index], y2[remain_indices])
-		width = numpy.maximum(0, xx2 - xx1 + 1)
-		height = numpy.maximum(0, yy2 - yy1 + 1)
-		iou = width * height / (areas[index] + areas[remain_indices] - width * height)
-		indices = indices[numpy.where(iou <= iou_limit)[0] + 1]
+def convert_bounding_box_to_rotated_rectangle(bounding_box : BoundingBox) -> RotatedRectangle:
+	x1, y1, x2, y2, angle = bounding_box
+	return ( x1, y1 ), ( int(x2 - x1), int(y2 - y1) ), angle
```
Comment (henryruhs) [inline facefusion/face_helper.py:141]: I don't like calculation in a return

File: facefusion/face_helper.py [hunk 3]
```diff
@@ -136,28 +136,14 @@ def convert_face_landmark_68_to_5(face_landmark_68 : FaceLandmark68) -> FaceLand
...
+
+
+def apply_nms(bounding_boxes : List[BoundingBox], face_scores : List[Score], score_threshold : float,  nms_threshold : float) -> Sequence[int]:
+	rotated_rectangles = list(map(convert_bounding_box_to_rotated_rectangle, bounding_boxes))
```
Comment (henryruhs) [inline facefusion/face_helper.py:145]: rotated_rectangle is a bad name, this are still bounding boxes

File: facefusion/face_helper.py [hunk 4]
```diff
@@ -136,28 +136,14 @@ def convert_face_landmark_68_to_5(face_landmark_68 : FaceLandmark68) -> FaceLand
...
-		keep_indices.append(index)
-		xx1 = numpy.maximum(x1[index], x1[remain_indices])
-		yy1 = numpy.maximum(y1[index], y1[remain_indices])
-		xx2 = numpy.minimum(x2[index], x2[remain_indices])
-		yy2 = numpy.minimum(y2[index], y2[remain_indices])
-		width = numpy.maximum(0, xx2 - xx1 + 1)
-		height = numpy.maximum(0, yy2 - yy1 + 1)
-		iou = width * height / (areas[index] + areas[remain_indices] - width * height)
-		indices = indices[numpy.where(iou <= iou_limit)[0] + 1]
+def convert_bounding_box_to_rotated_rectangle(bounding_box : BoundingBox) -> RotatedRectangle:
```
Comment (henryruhs) [inline facefusion/face_helper.py:139]: why is this not the format we store our bounding boxes in the first place?

File: facefusion/face_helper.py (hunk 3 above)
Comment (harisreedhar) [reply facefusion/face_helper.py:145]: cv2 calls it Rotated_Rect, any other suggestion? 
https://docs.opencv.org/3.4/db/dd6/classcv_1_1RotatedRect.html#ae1be388780b8d5faf450be18cbbf30f1

**Pull request merged**